src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -4255,6 +4255,62 @@ module Queries = struct
         Mina_lib.runtime_config mina
         |> Runtime_config.to_yojson |> Yojson.Safe.to_basic )
 
+  let get_epoch_ledgers ~mina breadcrumb =
+    let open Deferred.Result.Let_syntax in
+    let mina_config = Mina_lib.config mina in
+    let frontier_consensus_local_state = mina_config.consensus_local_state in
+    let consensus_constants =
+      mina_config.precomputed_values.consensus_constants
+    in
+    let consensus_state =
+      breadcrumb |> Transition_frontier.Breadcrumb.protocol_state
+      |> Mina_state.Protocol_state.consensus_state
+    in
+    let staking_epoch =
+      Consensus.Proof_of_stake.Data.Consensus_state.staking_epoch_data
+        consensus_state
+    in
+    let next_epoch =
+      Consensus.Proof_of_stake.Data.Consensus_state.next_epoch_data
+        consensus_state
+    in
+    let staking_ledger =
+      match
+        Consensus.Hooks.get_epoch_ledger ~constants:consensus_constants
```

Comment:
`breadcrumb` refers to the last block before stop-tx-slot.

If it's in the first `k` blocks of the epoch, whereas best tip is after `k` blocks in the epoch, there is a trouble that `get_epoch_ledger` would return `local_state.next_epoch_ledger` when the correct value is  `local_state.staking_epoch_ledger`.

Reply:
My reading of the code is: the epoch ledgers refer to the frontier root and not the best tip. From that perspective, `get_epoch_ledger` does indeed always return the correct ledger.

Reply:
I agree that epoch ledgers refer to the root.

In the case I'm describing root is in the current epoch `e`, so `local_state.next_epoch_ledger` refers to the next epoch `e+1` and the `get_epoch_ledger` as I understand would return `local_state.next_epoch_ledger` because the `is_finalized` condition would evaluate to false for the `breadcrumb`.